src/lib/y2network/widgets/wireless_essid.rb

```diff
@@ -110,21 +112,47 @@ def label
       end
 
       def handle
-        networks = essid_list
+        return unless scan_supported?
+
+        @update_widget&.update_essid_list(fetch_essid_list)
+        nil
+      end
+
+    private
+
+      IWLIST_PKG = "wireless-tools".freeze
+
+      def scan_supported?
+        return true if package_installed?
+
+        Yast::Popup.Error(
+          _("The package %s was not installed. It is needed in order to " \
+            "be able to scan the network") % IWLIST_PKG
+        )
+        false
+      end
+
+      # Require wireless-tools installation in order to be able to scan the
+      # wlan network (bsc#1112952, bsc#1168479)
+      #
+      # TODO: drop it when supported by wicked directly
+      def package_installed?
+        Yast::Stage.initial ||
+          Yast::Package.Installed(IWLIST_PKG) ||
+          Yast::Package.Install(IWLIST_PKG)
```

Comment:
the last part doesn't seem to be answer for ```package_installed?```

Reply:
i mean it sounds like check whether the package is installed but in fact it has sideeffect installing the package.

Reply:
Yep, I was not sure about the naming, and moving it from the method was an option but do not liked at all. I named firstly `ensure_package_installed`

Reply:
we could use `package_installed? || Yast::Package.Install(IWLIST_PKG)` if you prefer

Reply:
@mchf I will rename as `install_needed_packages`,  it is fine with you?

Reply:
yep